### PR TITLE
fix: add more logging to data-plane extensions (fixes #2122)

### LINKED ISSUE #2122: Add more logging to data-plane extensions
# Bug Report

## Observed Behavior

data-plane extensions are quiet comparing to control-plane ones.

## Expected Behavior

It should emit log about conditionally loaded modules and negotiated configurations.

## Steps to Reproduce

```
$ ./gradlew -p system-tests/e2e-transfer-test/runner clean test -DincludeTags="EndToEndTest" -PverboseTest --tests '*httpPushDataTransfer'
```

## Context Information

- Used version: main 93f53eb91
- OS: Ubuntu 20.04 x86_64
- OpenJDK 11.0.16+8-post-Ubuntu-0ubuntu122.04

### PR DESCRIPTION
**Pull request by iwasakims**
## What this PR changes/adds

This PR improves logging about conditionally loaded modules and negotiated configurations.

## Why it does that

Since it helps debugging and testing.

## Linked Issue(s)

Closes #2122

## Checklist

- [n/a] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [n/a] documented public classes/methods?
- [n/a] added/updated relevant documentation?
- [x] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java [hunk 1]
```diff
@@ -106,17 +115,20 @@ public CompletableFuture<StatusResult<Void>> transfer(DataSink sink, DataFlowReq
...
     @Override
     public void registerFactory(DataSourceFactory factory) {
         sourceFactories.add(factory);
+        monitor.info(() -> format("%s was registered: %s.", DataSourceFactory.class.getSimpleName(), factory.getClass().getSimpleName()));
     }
 
     @Override
     public void registerFactory(DataSinkFactory factory) {
         sinkFactories.add(factory);
+        monitor.info(() -> format("%s was registered: %s.", DataSinkFactory.class.getSimpleName(), factory.getClass().getSimpleName()));
```
Comment (jimmarino) [inline core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:131]: Please remove this statement as it is extraneous - the runtime will already emit a message that the extension was loaded, and it doesn't need to record a factory was registered.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java [hunk 2]
```diff
@@ -106,17 +115,20 @@ public CompletableFuture<StatusResult<Void>> transfer(DataSink sink, DataFlowReq
...
         }
         var source = sourceFactory.createSource(request);
+        monitor.debug(() -> format("transferring from %s to %s.", source, sink));
         return sink.transfer(source);
     }
 
     @Override
     public void registerFactory(DataSourceFactory factory) {
         sourceFactories.add(factory);
+        monitor.info(() -> format("%s was registered: %s.", DataSourceFactory.class.getSimpleName(), factory.getClass().getSimpleName()));
```
Comment (jimmarino) [inline core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:125]: Please remove this statement as it is extraneous - the runtime will already emit a message that the extension was loaded, and it doesn't need to record a factory was registered.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java [hunk 3]
```diff
@@ -129,11 +132,13 @@ public void shutdown() {
 
     @NotNull
     private DataPlaneStore registerStore(ServiceExtensionContext context) {
+        var monitor = context.getMonitor();
         if (store != null) {
             return store;
         }
         var inMemoryStore = new InMemoryDataPlaneStore(IN_MEMORY_STORE_CAPACITY);
         context.registerService(DataPlaneStore.class, inMemoryStore);
+        monitor.info(() -> format("Using %s with capacity=%s.", DataPlaneStore.class.getSimpleName(), IN_MEMORY_STORE_CAPACITY));
```
Comment (jimmarino) [inline core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java:141]: This should not be logged at info level. Please convert to debug level.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java [hunk 4]
```diff
@@ -77,7 +78,8 @@ public String name() {
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var pipelineService = new PipelineServiceImpl();
+        var monitor = context.getMonitor();
```
Comment (jimmarino) [inline core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java:81]: Can you please change all of the log messages below from info to debug and consolidate them into one message?

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java [hunk 5]
```diff
@@ -66,16 +68,20 @@ public String name() {
 
     @Override
     public void initialize(ServiceExtensionContext context) {
+        var monitor = context.getMonitor();
+
         DataPlaneTransferClient client;
         if (dataPlaneManager != null) {
             // Data plane manager is embedded in the current runtime
             client = new EmbeddedDataPlaneTransferClient(dataPlaneManager);
+            monitor.info(() -> format("Using %s.", EmbeddedDataPlaneTransferClient.class.getSimpleName()));
```
Comment (jimmarino) [inline extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java:77]: Please remove this message, it does not provide useful operation information

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java [hunk 6]
```diff
@@ -66,16 +68,20 @@ public String name() {
...
         } else {
             Objects.requireNonNull(okHttpClient, "If no DataPlaneManager is embedded, a OkHttpClient instance must be provided");
             Objects.requireNonNull(retryPolicy, "If no DataPlaneManager is embedded, a RetryPolicy instance must be provided");
             Objects.requireNonNull(selectorClient, "If no DataPlaneManager is embedded, a DataPlaneSelector instance must be provided");
             var selectionStrategy = context.getSetting(DPF_SELECTOR_STRATEGY, "random");
             client = new RemoteDataPlaneTransferClient(okHttpClient, selectorClient, selectionStrategy, retryPolicy, context.getTypeManager().getMapper());
+            monitor.info(() -> format("Using %s with selectionStratey=%s.", RemoteDataPlaneTransferClient.class.getSimpleName(), selectionStrategy));
```
Comment (jimmarino) [inline extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java:84]: Please change this to debug level

**Review by jimmarino — CHANGES_REQUESTED** [review]
Thanks for the submission! Comments inline

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 1 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:131]: For example, I can not see the log about loading DataPlaneHttpExtension (which registers HttpDataSourceFactory and HttpDataSinkFactory) in the sample log like:

```
$ ./gradlew -p system-tests/e2e-transfer-test/runner clean test -DincludeTags="EndToEndTest" -PverboseTest --tests '*httpPushDataTransfer'
```

In addition, extensions can conditionally switch and register source/sink based on the condition. Logging which source/sink is registered is useful for finding issues even after the process is crashed. @jimmarino

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java (hunk 5 above)
Comment (iwasakims) [reply extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java:77]: It should be useful to detect unintensional setup in the first place [as DataPlaneInstanceClientExtension  do](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/93f53eb917fd5ffc0670a909bb05dd92fd542c42/extensions/data-plane-selector/data-plane-selector-client/src/main/java/org/eclipse/dataspaceconnector/dataplane/selector/DataPlaneInstanceClientExtension.java#L58-L67).

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java (hunk 4 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java:81]: > Can you please change all of the log messages below from info to debug and consolidate them into one message?

I think those one-time message on staring up are worth for INFO level. It is quite useful to show setup in the startup message for finding issues like loading unintentional configuration files on operation.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 1 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:131]: > For example, I can not see the log about loading DataPlaneHttpExtension

Sorry, it was not true.

```
    [provider-data-plane] INFO 2022-10-23T15:41:08.144923 Initialized Data Plane HTTP
```

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 2 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:125]: I still want debug logging for extensions dynamically switching  and registering source/sink based on the condition.

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java (hunk 6 above)
Comment (iwasakims) [reply extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java]: updated log lovel.

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java (hunk 5 above)
Comment (iwasakims) [reply extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java]: updated log level.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java (hunk 4 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java:81]: I updated the log level and consolidated the message.

**Comment by iwasakims** [comment]
I reduced the log level based on your suggestion as a starting point. @jimmarino

**Comment by paullatzelsperger** [comment]
@iwasakims I removed myself as a reviewer, since @jimmarino is already on it.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 2 above)
Comment (jimmarino) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java]: We explicitly designed extensions not to be dynamic like OSGI. The `PipelineService` is also designed not to be dynamic.

An event for the loaded extension will already be emitted, so doing so for the factory offers no value. Please remove it.

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java (hunk 5 above)
Comment (jimmarino) [reply extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java]: Can you either remove this or change the message emitted to something more useful? If a message is emitted, it should be a debug message with something like "Using embedded Data Plane", not the classname. Thx.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 2 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java]: removed.

File: extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java (hunk 5 above)
Comment (iwasakims) [reply extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/DataPlaneTransferClientExtension.java]: I updated the message.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java [hunk 7]
```diff
@@ -106,6 +115,7 @@ public CompletableFuture<StatusResult<Void>> transfer(DataSink sink, DataFlowReq
             return noSourceFactory(request);
         }
         var source = sourceFactory.createSource(request);
+        monitor.debug(() -> format("transferring from %s to %s.", source, sink));
```
Comment (jimmarino) [inline core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java:118]: Can you capitalize the emessage?

**Review by jimmarino — CHANGES_REQUESTED** [review]
Sorry, just a few more capitalizations that I didn't catch in the first pass. Thanks.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java (hunk 7 above)
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java]: I updated the messages. Thanks, @jimmarino.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java [hunk 8]
```diff
@@ -96,6 +104,7 @@ public CompletableFuture<StatusResult<Void>> transfer(DataSource source, DataFlo
             return noSinkFactory(request);
         }
         var sink = sinkFactory.createSink(request);
+        monitor.debug(() -> format("transferring from %s to %s.", source, sink));
```
[the remark this answers is not in the document]
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java]: fixed.

File: core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java [hunk 9]
```diff
@@ -86,6 +93,7 @@ public CompletableFuture<StatusResult<Void>> transfer(DataFlowRequest request) {
         }
         var source = sourceFactory.createSource(request);
         var sink = sinkFactory.createSink(request);
+        monitor.debug(() -> format("transferring from %s to %s.", source, sink));
```
[the remark this answers is not in the document]
Comment (iwasakims) [reply core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java]: fixed.

**Review by jimmarino — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**